core/primitives/src/shard_layout/mod.rs

```diff
@@ -328,6 +358,11 @@ impl ShardLayout {
     /// Returns all the shards from the previous shard layout that were
     /// split into multiple shards in this shard layout.
     pub fn get_split_parent_shard_ids(&self) -> BTreeSet<ShardId> {
+        // V3 doesn't store shards which weren't split in the map
+        if let ShardLayout::V3(v3) = self {
+            return v3.shards_split_map.keys().cloned().collect();
```

Comment:
That returns previous shard splits as well, not just the most recent one, right? But the semantics of `get_split_parent_shard_ids` is to only return parents from the most recent shard split.